daemon/policy.go

```diff
@@ -302,6 +304,79 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (newRev uint
 	return newRev, nil
 }
 
+type policyAddQueueOptions struct {
+	rules   policyAPI.Rules
+	opts    *AddOptions
+	context *policyQueueContext
+	reFetch bool
+}
+
+type policyQueueContext struct {
+	policyRevision uint64
+	err            error
+	done           chan bool
+	totalTime      spanstat.SpanStat
+	queueTime      spanstat.SpanStat
+}
+
+var policyAddQueueRequests = make(chan policyAddQueueOptions, 100)
```

Comment:
Make this a variable of the Daemon so we can unit test this in parallel. We don't want to introduce more global variables.